### DISCUSSION TITLE
Create email hook to send emails using platform.sh

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [directus/directus#4143] [Archive]
Hello, 

We have created a Directus instance using platform.sh one-click deployment, and it is working all fine.

Following the almost template to send emails using hooks: https://gist.github.com/juancarlosjr97/8078b0cf9055669ce993edbb15812f64 which works on Dockerised Directus locally or cloud-based passing the SMTP configuration. 

When adding the same configuration to Directus from platform.sh we encountered the issue that neither `env` or `process.env` return the following variables:

- EMAIL_SMTP_SECURE
- EMAIL_SMTP_USER
- EMAIL_SMTP_PASSWORD

If those variables are not available or set, how the Directus created using `platform.sh` without me passing the SMTP configuration it is sending emails successfully to reset password?

Environment:
* Platform.sh
* Directus rc.34

Thank you

**Comment by juanitourquiza** [comment]
Hi,

Thanks for your help @juancarlosjr97 
I don't need these variables in platform.sh

This is the code for sent the email:

```
module.exports = (router, { env, database }) => {
  router.get("/", async (req, res) => {
    const { Liquid } = require("liquidjs");
    const nodemailer = require("nodemailer");
    const path = require("path");

    const transporter = nodemailer.createTransport({
      host: env.EMAIL_SMTP_HOST,
      port: env.EMAIL_SMTP_PORT,
    });

    transporter.sendMail({
      from: env.EMAIL_FROM,
      to: "juanitourquiza@gmail.com",
      html: "kkkk",
      subject: `Test - Email`,
    });    
    res.send("");
  });
};
```

![image](https://user-images.githubusercontent.com/332299/108301596-58802100-7170-11eb-850b-9a618edb27b6.png)

**Reply by juancarlosjr97** [reply]
I am glad we found a solution @juanitourquiza :)
